astronomer/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any, Optional, Union
+
+from airflow import AirflowException
+from airflow.providers.microsoft.azure.hooks.data_factory import AzureDataFactoryHook
+from asgiref.sync import sync_to_async
+from azure.identity.aio import ClientSecretCredential, DefaultAzureCredential
+from azure.mgmt.datafactory.aio import DataFactoryManagementClient
+from azure.mgmt.datafactory.models import PipelineRun
+
+Credentials = Union[ClientSecretCredential, DefaultAzureCredential]
+
+
+class AzureDataFactoryHookAsync(AzureDataFactoryHook):
+    """
+    An Async Hook connects to Azure DataFactory to perform pipeline operations
+    :param azure_data_factory_conn_id: The :ref:`Azure Data Factory connection id<howto/connection:adf>`.
```

Comment:
Did you check whether  the docs got build correctly for this? I think we need to give a carriage return after the  end of description